File: 49_Group Anagrams.java
   import java.util.*;

   /**
    * 49_Group Anagrams
    */
   class Solution {
      public List<List<String>> groupAnagrams(String[] strs) {
         HashMap<String, List<String>> anagrams = new HashMap<>();
         for (String s : strs) {
            char[] ca = s.toCharArray();
            Arrays.sort(ca);
            String newStr = new String(ca);
            if (!anagrams.containsKey(newStr)) {
               List<String> anas = new ArrayList<>();
               anas.add(s);
               anagrams.put(newStr, anas);
            } else {
               anagrams.get(newStr).add(s);
            }
         }
         return new ArrayList<>(anagrams.values());
      }
   }

File: Blind_75/49. Group Anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        if len(strs) == 1:
            return [strs]

        res = {}

        for string in strs:
            sorted_string = str(sorted(string))
            if sorted_string in res.keys():
                res[sorted_string].append(string)
            else:
                res[sorted_string] = [string]

        return list(res.values())